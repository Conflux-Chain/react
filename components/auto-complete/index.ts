import AutoComplete, { useAutoCompleteHandle } from './auto-complete'
export { useAutoCompleteHandle }
export default AutoComplete
