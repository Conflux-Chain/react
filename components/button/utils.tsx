import React, { ReactNode } from 'react'
import { NormalSizes } from '../utils/prop-types'
import ButtonIcon from './button-icon'
import { ButtonProps } from './button'
import { ButtonGroupConfig } from '../button-group/button-group-context'

export const getButtonChildrenWithIcon = (
  loading: boolean,
  auto: boolean,
  size: NormalSizes,
  children: ReactNode,
  icons: {
    icon?: React.ReactNode
    iconRight?: React.ReactNode
  },
  isTag: boolean,
) => {
  const { icon, iconRight } = icons
  const hasIcon = icon || iconRight
  const isRight = Boolean(iconRight)
  const paddingForAutoMode =
    auto || size === 'mini' || isTag
      ? `calc(var(--zeit-ui-button-height) / 2 + var(--zeit-ui-button-padding) * .5)`
      : 0
  if (!hasIcon) return <div className={`text ${loading ? 'hidden' : ''}`}>{children}</div>
  return (
    <>
      <ButtonIcon isRight={isRight}>{hasIcon}</ButtonIcon>
      <div className={`text ${isRight ? 'right' : 'left'} ${loading ? 'hidden' : ''}`}>
        {children}
        <style jsx>{`
          .left {
            padding-left: ${paddingForAutoMode};
          }
          .right {
            padding-right: ${paddingForAutoMode};
          }
        `}</style>
      </div>
    </>
  )
}

export const filterPropsWithGroup = (
  props: React.PropsWithChildren<ButtonProps>,
  config: ButtonGroupConfig,
): ButtonProps => {
  if (!config.isButtonGroup) return props
  return {
    ...props,
    auto: true,
    shadow: false,
    ghost: config.ghost || props.ghost,
    size: config.size || props.size,
    variant: config.variant || props.variant,
    color: config.color || props.color,
    disabled: config.disabled || props.disabled,
  }
}
