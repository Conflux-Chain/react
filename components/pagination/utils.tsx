export const getPageCount = (total: number, pageSize: number) => {
  return Math.floor((total - 1) / pageSize) + 1
}
