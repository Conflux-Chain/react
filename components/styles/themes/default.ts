import { ZeitUIThemes, ZeitUIThemesPalette, ZeitUIThemesExpressiveness } from './index'
import { defaultFont, defaultBreakpoints, defaultLayout } from './shared'

export const palette: ZeitUIThemesPalette = {
  /* eslint-disable @typescript-eslint/camelcase */
  accents_1: '#fafafa',
  accents_2: '#eaeaea',
  accents_3: '#999',
  accents_4: '#888',
  accents_5: '#666',
  accents_6: '#444',
  accents_7: '#333',
  accents_8: '#111',
  /* eslint-enable camelcase */
  background: '#fff',
  foreground: '#000',
  selection: '#79ffe1',
  secondary: '#666',
  code: '#f81ce5',
  border: '#eaeaea',
  error: '#e15c56',
  errorLight: '#ff8585',
  errorLighter: '#f7d4d6',
  errorDark: '#b34a50',
  success: '#58f17a',
  successLight: '#8afb88',
  successLighter: '#d3e5ff',
  successDark: '#61be8f',
  warning: '#ffc438',
  warningLight: '#ffde40',
  warningLighter: '#ffefcf',
  warningDark: '#f48b4f',
  cyan: '#50e3c2',
  cyanLighter: '#aaffec',
  cyanLight: '#79ffe1',
  cyanDark: '#29bc9b',
  violet: '#7928ca',
  violetLighter: '#e3d7fc',
  violetLight: '#8a63d2',
  violetDark: '#4c2889',
  purple: '#f81ce5',
  alert: '#ff0080',
  magenta: '#eb367f',
  link: '#0070f3',

  // conflux ui
  cTheme0: '#e8f4ff',
  cTheme1: '#cce5ff',
  cTheme2: '#c0d8fc',
  cTheme3: '#a9bde6',
  cTheme4: '#6d8bc8',
  cTheme5: '#0054fe',
  cTheme6: '#0433dc',
  cTheme7: '#0626ae',
  cBlack0: '#282D30',
  cGray0: '#f9fafb',
  cGray1: '#f1f3f6',
  cGray2: '#ebeced',
  cGray3: '#dbdde0',
  cGray4: '#aaafbb',
  cGray5: '#9b9eac',
  cGray6: '#444',
  cWhite0: '#fff',
  brandLight: '#c0d8fc',
  brand: '#0054fe',
  brandDark: '#0626ae',
}

export const expressiveness: ZeitUIThemesExpressiveness = {
  linkStyle: 'none',
  linkHoverStyle: 'none',
  dropdownBoxShadow: '0 4px 4px 0 rgba(0, 0, 0, 0.02)',
  scrollerStart: 'rgba(255, 255, 255, 1)',
  scrollerEnd: 'rgba(255, 255, 255, 0)',
  shadowSmall: '0 5px 10px rgba(0, 0, 0, 0.12)',
  shadowMedium: '0 8px 30px rgba(0, 0, 0, 0.12)',
  shadowLarge: '0 30px 60px rgba(0, 0, 0, 0.12)',
  portalOpacity: 0.25,

  // radius
  R0: '0',
  R1: '2px',
  R2: '4px',
  R3: '8px',
  R4: '12px',
  R5: '50%',

  // drop shadows
  D0: 'none',
  D1: '0px 0px 2px rgba(0, 0, 0, 0.08)',
  D2: '0px 4px 8px rgba(0, 0, 0, 0.08)',
  D3: '0px 4px 8px rgba(0, 0, 0, 0.16)',
  D4: '0px 8px 16px rgba(0, 0, 0, 0.24)',
}

export const font = defaultFont

export const breakpoints = defaultBreakpoints

export const layout = defaultLayout

export const themes: ZeitUIThemes = {
  type: 'light',
  font,
  layout,
  palette,
  breakpoints,
  expressiveness,
}

export default themes
