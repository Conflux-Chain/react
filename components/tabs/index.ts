import Tabs from './tabs'
import TabsItem from './tabs-item'
import useTabsHandle from './use-tabs-handle'

Tabs.Item = TabsItem
Tabs.Tab = TabsItem
Tabs.useTabsHandle = useTabsHandle

export default Tabs
