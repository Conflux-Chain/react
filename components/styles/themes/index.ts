import { ThemeTypes } from '../../utils/prop-types'

export interface ZeitUIThemesPalette {
  /* eslint-disable @typescript-eslint/camelcase */
  accents_1: string
  accents_2: string
  accents_3: string
  accents_4: string
  accents_5: string
  accents_6: string
  accents_7: string
  accents_8: string
  /* eslint-enable camelcase */
  background: string
  foreground: string
  selection: string
  secondary: string
  code: string
  border: string
  success: string
  successLighter: string
  successLight: string
  successDark: string
  error: string
  errorLighter: string
  errorLight: string
  errorDark: string
  warning: string
  warningLighter: string
  warningLight: string
  warningDark: string
  cyan: string
  cyanLighter: string
  cyanLight: string
  cyanDark: string
  violet: string
  violetLighter: string
  violetLight: string
  violetDark: string
  link: string
  purple: string
  magenta: string
  alert: string

  // conflux
  cThem0: string
  cThem1: string
  cThem2: string
  cThem3: string
  cThem4: string
  cThem5: string
  cThem6: string
  cThem7: string
  cBlack0: string
  cGray0: string
  cGray1: string
  cGray2: string
  cGray3: string
  cGray4: string
  cGray5: string
  cGray6: string
  cWhite0: string
  brandLight: string
  brand: string
  brandDark: string
}

export interface ZeitUIThemesExpressiveness {
  linkStyle: string
  linkHoverStyle: string
  dropdownBoxShadow: string
  scrollerStart: string
  scrollerEnd: string
  shadowSmall: string
  shadowMedium: string
  shadowLarge: string
  portalOpacity: number

  // radius
  R0: string
  R1: string
  R2: string
  R3: string
  R4: string
  R5: string

  // drop shadows
  D0: string
  D1: string
  D2: string
  D3: string
  D4: string
}

export interface ZeitUIThemesLayout {
  gap: string
  gapNegative: string
  gapHalf: string
  gapHalfNegative: string
  gapQuarter: string
  gapQuarterNegative: string
  pageMargin: string
  pageWidth: string
  pageWidthWithMargin: string
  breakpointMobile: string
  breakpointTablet: string
  radius: string
}

export interface ZeitUIThemesFont {
  sans: string
  mono: string
  circular: string
  pingfang: string
}

export interface BreakpointsItem {
  min: string
  max: string
}

export interface ZeitUIThemesBreakpoints {
  xs: BreakpointsItem
  sm: BreakpointsItem
  md: BreakpointsItem
  lg: BreakpointsItem
  xl: BreakpointsItem
}

export interface ZeitUIThemes {
  type: ThemeTypes
  font: ZeitUIThemesFont
  layout: ZeitUIThemesLayout
  palette: ZeitUIThemesPalette
  breakpoints: ZeitUIThemesBreakpoints
  expressiveness: ZeitUIThemesExpressiveness
}
