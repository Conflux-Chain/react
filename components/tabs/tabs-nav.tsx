import React, { useMemo } from 'react'
import { TabVariant } from 'components/utils/prop-types'
import { defaultGetColor } from './style'
import useTheme from '../styles/use-theme'

export interface NavCptProps {
  variant: TabVariant
  disabled: boolean
  active: boolean
  label: string | React.ReactNode
}

const Nav: React.FC<NavCptProps> = ({ label, variant, disabled, active }) => {
  const { palette, layout, expressiveness } = useTheme()
  const colors = useMemo(() => defaultGetColor(palette, variant, disabled, active), [
    variant,
    disabled,
    active,
  ])
  return (
    <div className="nav">
      <div className="label">{label}</div>
      {variant === 'line' ? <div className="bottom"></div> : null}
      <style jsx>
        {`
          .nav {
            display: flex;
            flex-direction: column;
            position: relative;
          }
          .label {
            cursor: ${disabled ? 'not-allowed' : ''};
            font-weight: ${active ? '500' : '400'};
            white-space: nowrap;
            line-height: 22px;
            padding: ${layout.gapHalf} ${layout.gap};
            border-radius: ${expressiveness.R2} ${expressiveness.R2} 0px 0px;
            text-align: center;
            color: ${colors.color};
            background-color: ${colors.bgColor};
          }
          .label:hover {
            color: ${colors.hoverColor};
            background-color: ${colors.hoverBgColor};
          }
          .bottom {
            background-color: ${palette.cTheme5};
            transition: all 200ms ease;
            opacity: ${active ? '1' : '0'};
            transform: scale(${active ? '1' : '0.75'});
            height: 4px;
            width: 100%;
          }
        `}
      </style>
    </div>
  )
}

export default Nav
