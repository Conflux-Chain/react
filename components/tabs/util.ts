import { TabVarient, TabStatus } from '../utils/prop-types'
import { ZeitUIThemesPalette } from 'components/styles/themes'
const stack: TabStatus[] = ['disabled', 'active', 'hover', 'default']

export type StatusMap = {
  [key in TabStatus]?: boolean
}

export type ColorFN = (palette: ZeitUIThemesPalette, varient: TabVarient, status: TabStatus) => any

export const defaultGetColor: any = (
  palette: ZeitUIThemesPalette,
  varient: TabVarient,
  status: TabStatus,
) => {
  if (status === 'active' || status === 'hover') {
    return {
      color: palette.cTheme5,
      background:
        varient === 'line' ? '' : status === 'active' ? palette.cTheme0 : palette.cNeutral1,
    }
  } else if (status === 'default') {
    return {
      color: palette.cNeutral6,
      background: varient === 'solid' ? palette.cNeutral1 : '',
    }
  } else if (status === 'disabled') {
    return {
      color: palette.cNeutral5,
      background: varient === 'solid' ? palette.cNeutral0 : '',
    }
  }
}

export function reduceStatus(props: StatusMap): TabStatus {
  for (let i = 0; i <= stack.length; i++) {
    const s = stack[i]
    if (props[s]) {
      return s
    }
  }
  return 'default'
}
