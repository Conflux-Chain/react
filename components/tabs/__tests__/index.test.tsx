import React, { useEffect, useState } from 'react'
import { nativeEvent } from 'tests/utils'
import { mount, ReactWrapper } from 'enzyme'
import { nav } from '../tabs-nav'
import { reduceStatus, defaultGetColor } from '../util'
import { Tabs } from 'components'
import { palette } from '../../styles/themes/default'

describe('Tabs', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Tabs>
        <Tabs.Item label="label1" value="1">
          1
        </Tabs.Item>
        <Tabs.Item label="label2" value="2">
          2
        </Tabs.Item>
      </Tabs>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should trigger events when tab changed', () => {
    let value = ''
    const changeHandler = jest.fn().mockImplementation(val => (value = val))
    const wrapper = mount(
      <Tabs initialValue="1" onChange={changeHandler}>
        <Tabs.Item label="label1" value="1">
          1
        </Tabs.Item>
        <Tabs.Item label="label2" value="2">
          2
        </Tabs.Item>
      </Tabs>,
    )

    wrapper.find('header').find('.tab').at(1).simulate('click', nativeEvent)
    expect(changeHandler).toHaveBeenCalled()
    expect(value).toBe('2')
  })

  it('should change style when hover header', () => {
    const wrapper = mount(
      <Tabs initialValue="1">
        <Tabs.Item label="label1" value="1">
          1
        </Tabs.Item>
        <Tabs.Item label="label2" value="2">
          2
        </Tabs.Item>
      </Tabs>,
    )
    wrapper.find('header').find('.nav').at(1).simulate('mouseEnter', nativeEvent)
    expect(wrapper.find('header').find('.label').at(1).get(0).props.style).toHaveProperty(
      'color',
      '#0054fe',
    )
    wrapper.find('header').find('.nav').at(1).simulate('mouseLeave', nativeEvent)
    expect(wrapper.find('header').find('.label').at(1).get(0).props.style).toHaveProperty(
      'color',
      '#444',
    )
  })

  it('can set tabs dynamically', () => {
    function App() {
      const [items, setItems] = useState([
        { label: 'label1', value: '1' },
        { label: 'label2', value: '2' },
      ])
      useEffect(() => {
        setItems([{ label: 'label11', value: '1' }])
      }, [])
      return (
        <Tabs initialValue="1">
          {items.map(({ label, value }) => {
            return (
              <Tabs.Item key={value} label={label} value={value}>
                {label}
              </Tabs.Item>
            )
          })}
        </Tabs>
      )
    }
    const wrapper = mount(<App />)
    const label = (wrapper.find('header').find('.label').at(0).props().children as Array<any>)[0]
    expect(label).toBe('label11')
    wrapper.unmount()
  })

  it('can be controlled', () => {
    const controlled = mount(
      <Tabs value="1">
        <Tabs.Item label="label1" value="1">
          1
        </Tabs.Item>
        <Tabs.Item label="label2" value="2">
          2
        </Tabs.Item>
      </Tabs>,
    )
    const uncontrolled = mount(
      <Tabs initialValue="1">
        <Tabs.Item label="label1" value="1">
          1
        </Tabs.Item>
        <Tabs.Item label="label2" value="2">
          2
        </Tabs.Item>
      </Tabs>,
    )
    // const a = wrapper.find(Tabs.Item).at(0).html()
    uncontrolled.find('header').find('.tab').at(1).simulate('click', nativeEvent)
    controlled.find('header').find('.tab').at(1).simulate('click', nativeEvent)
    const getActiveLabel = (wrapper: ReactWrapper) =>
      wrapper
        .find('header')
        .findWhere(n => n.hasClass('active'))
        .find('.label')
        .childAt(0)
        .html()
    expect(getActiveLabel(uncontrolled)).toBe('label2')
    expect(getActiveLabel(controlled)).toBe('label1')
  })

  it('should ignore events when tab disabled', () => {
    const changeHandler = jest.fn()
    const wrapper = mount(
      <Tabs initialValue="1" onChange={changeHandler}>
        <Tabs.Item label="label1" value="1">
          1
        </Tabs.Item>
        <Tabs.Item label="label2" value="2" disabled>
          2
        </Tabs.Item>
      </Tabs>,
    )

    wrapper.find('header').find('.tab').at(1).simulate('click', nativeEvent)
    expect(changeHandler).not.toHaveBeenCalled()
  })

  it('should return default nav', () => {
    expect(() => nav({})).not.toThrow()
  })

  it('should be able to update tab-item props', () => {
    function App() {
      const [label1, setLabel1] = useState('label1')
      useEffect(() => {
        setLabel1('label11')
      }, [])
      return (
        <Tabs value="1" showDivider={true}>
          <Tabs.Item label={label1} value="1">
            1
          </Tabs.Item>
          <Tabs.Item label="label2" value="2" disabled>
            2
          </Tabs.Item>
        </Tabs>
      )
    }
    const wrapper = mount(<App />)
    const label = wrapper
      .find('header')
      .findWhere(n => n.hasClass('active'))
      .find('.label')
      .childAt(0)
      .html()
    expect(label).toBe('label11')
  })
})

describe('useImperative', () => {
  it('should work with useImperative', () => {
    const { useTabsHandle } = Tabs
    function App() {
      const { ref, setCurrentTab, getCurrentTab } = useTabsHandle()
      useEffect(() => {
        getCurrentTab()
        setCurrentTab('2')
      }, [])
      return (
        <Tabs ref={ref}>
          <Tabs.Item label="label1" value="1">
            1
          </Tabs.Item>
          <Tabs.Item label="label2" value="2" disabled>
            2
          </Tabs.Item>
        </Tabs>
      )
    }

    const wrapper = mount(<App />)
    const label = wrapper
      .find('header')
      .findWhere(n => n.hasClass('active'))
      .find('.label')
      .childAt(0)
      .html()
    expect(label).toBe('label2')
  })
})

describe('utils', () => {
  it('should cover a unreachable brach due to a ts bug', () => {
    reduceStatus({})
  })

  it('should return corrent colors with virant and status combination', () => {
    const status = ['disabled', 'active', 'hover', 'default']
    const varients = ['line', 'solid']
    const colors: string[] = []
    status.forEach(s => {
      varients.forEach(v => {
        colors.push(defaultGetColor(palette, v, s))
      })
    })
    expect(colors).toMatchSnapshot()
  })
})
