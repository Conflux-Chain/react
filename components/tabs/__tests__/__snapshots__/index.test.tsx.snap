// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should change style when hover header 1`] = `
"<div class=\\"tabs \\"><header><div class=\\"tab active\\" role=\\"button\\"><div class=\\"nav\\"><div class=\\"label\\">label1</div><div class=\\"bottom\\"></div><style>
          .nav {
            display: flex;
            flex-direction: column;
            position: relative;
          }
          .label {
            cursor: ;
            font-weight: 500;
            white-space: nowrap;
            line-height: 22px;
            padding: 8px 16px;
            border-radius: 4px 4px 0px 0px;
            text-align: center;
            color: #0054fe;
            background-color: ;
          }
          .label:hover {
            color: #0054fe;
            background-color: ;
          }
          .bottom {
            background-color: #0054fe;
            transition: all 200ms ease;
            opacity: 1;
            transform: scale(1);
            height: 4px;
            width: 100%;
          }
        </style></div></div><div class=\\"tab \\" role=\\"button\\"><div class=\\"nav\\"><div class=\\"label\\">label2</div><div class=\\"bottom\\"></div><style>
          .nav {
            display: flex;
            flex-direction: column;
            position: relative;
          }
          .label {
            cursor: ;
            font-weight: 400;
            white-space: nowrap;
            line-height: 22px;
            padding: 8px 16px;
            border-radius: 4px 4px 0px 0px;
            text-align: center;
            color: #444;
            background-color: ;
          }
          .label:hover {
            color: #0054fe;
            background-color: ;
          }
          .bottom {
            background-color: #0054fe;
            transition: all 200ms ease;
            opacity: 0;
            transform: scale(0.75);
            height: 4px;
            width: 100%;
          }
        </style></div></div></header><div class=\\"content\\">1</div><style>
            header {
              display: flex;
              flex-wrap: nowrap;
              overflow-x: auto;
              align-items: center;
              border-bottom: ;
            }
            ::-webkit-scrollbar {
              display: none;
            }
            .content {
              padding-top: 0.625rem;
            }

            .tab {
              cursor: pointer;
              outline: 0;
              text-transform: capitalize;
              font-size: 1rem;
              margin: 0 1px;
              user-select: none;
              display: flex;
              align-items: center;
              line-height: 1.25rem;
              position: relative;
              font-variant-numeric: tabular-nums;
            }
            .tab:first-of-type {
              margin-left: 0;
            }
          </style></div>"
`;

exports[`Tabs should change style when hover header 2`] = `
"<div class=\\"tabs \\"><header><div class=\\"tab active\\" role=\\"button\\"><div class=\\"nav\\"><div class=\\"label\\">label1</div><div class=\\"bottom\\"></div><style>
          .nav {
            display: flex;
            flex-direction: column;
            position: relative;
          }
          .label {
            cursor: ;
            font-weight: 500;
            white-space: nowrap;
            line-height: 22px;
            padding: 8px 16px;
            border-radius: 4px 4px 0px 0px;
            text-align: center;
            color: #0054fe;
            background-color: ;
          }
          .label:hover {
            color: #0054fe;
            background-color: ;
          }
          .bottom {
            background-color: #0054fe;
            transition: all 200ms ease;
            opacity: 1;
            transform: scale(1);
            height: 4px;
            width: 100%;
          }
        </style></div></div><div class=\\"tab \\" role=\\"button\\"><div class=\\"nav\\"><div class=\\"label\\">label2</div><div class=\\"bottom\\"></div><style>
          .nav {
            display: flex;
            flex-direction: column;
            position: relative;
          }
          .label {
            cursor: ;
            font-weight: 400;
            white-space: nowrap;
            line-height: 22px;
            padding: 8px 16px;
            border-radius: 4px 4px 0px 0px;
            text-align: center;
            color: #444;
            background-color: ;
          }
          .label:hover {
            color: #0054fe;
            background-color: ;
          }
          .bottom {
            background-color: #0054fe;
            transition: all 200ms ease;
            opacity: 0;
            transform: scale(0.75);
            height: 4px;
            width: 100%;
          }
        </style></div></div></header><div class=\\"content\\">1</div><style>
            header {
              display: flex;
              flex-wrap: nowrap;
              overflow-x: auto;
              align-items: center;
              border-bottom: ;
            }
            ::-webkit-scrollbar {
              display: none;
            }
            .content {
              padding-top: 0.625rem;
            }

            .tab {
              cursor: pointer;
              outline: 0;
              text-transform: capitalize;
              font-size: 1rem;
              margin: 0 1px;
              user-select: none;
              display: flex;
              align-items: center;
              line-height: 1.25rem;
              position: relative;
              font-variant-numeric: tabular-nums;
            }
            .tab:first-of-type {
              margin-left: 0;
            }
          </style></div>"
`;

exports[`Tabs should render correctly 1`] = `
"<div><div class=\\"tabs \\"><header><div class=\\"tab \\" role=\\"button\\"><div class=\\"nav\\"><div class=\\"label\\">label1</div><div class=\\"bottom\\"></div><style>
          .nav {
            display: flex;
            flex-direction: column;
            position: relative;
          }
          .label {
            cursor: ;
            font-weight: 400;
            white-space: nowrap;
            line-height: 22px;
            padding: 8px 16px;
            border-radius: 4px 4px 0px 0px;
            text-align: center;
            color: #444;
            background-color: ;
          }
          .label:hover {
            color: #0054fe;
            background-color: ;
          }
          .bottom {
            background-color: #0054fe;
            transition: all 200ms ease;
            opacity: 0;
            transform: scale(0.75);
            height: 4px;
            width: 100%;
          }
        </style></div></div><div class=\\"tab \\" role=\\"button\\"><div class=\\"nav\\"><div class=\\"label\\">label2</div><div class=\\"bottom\\"></div><style>
          .nav {
            display: flex;
            flex-direction: column;
            position: relative;
          }
          .label {
            cursor: ;
            font-weight: 400;
            white-space: nowrap;
            line-height: 22px;
            padding: 8px 16px;
            border-radius: 4px 4px 0px 0px;
            text-align: center;
            color: #444;
            background-color: ;
          }
          .label:hover {
            color: #0054fe;
            background-color: ;
          }
          .bottom {
            background-color: #0054fe;
            transition: all 200ms ease;
            opacity: 0;
            transform: scale(0.75);
            height: 4px;
            width: 100%;
          }
        </style></div></div></header><div class=\\"content\\"></div><style>
            header {
              display: flex;
              flex-wrap: nowrap;
              overflow-x: auto;
              align-items: center;
              border-bottom: ;
            }
            ::-webkit-scrollbar {
              display: none;
            }
            .content {
              padding-top: 0.625rem;
            }

            .tab {
              cursor: pointer;
              outline: 0;
              text-transform: capitalize;
              font-size: 1rem;
              margin: 0 1px;
              user-select: none;
              display: flex;
              align-items: center;
              line-height: 1.25rem;
              position: relative;
              font-variant-numeric: tabular-nums;
            }
            .tab:first-of-type {
              margin-left: 0;
            }
          </style></div><div class=\\"tabs \\"><header><div class=\\"tab \\" role=\\"button\\"><div class=\\"nav\\"><div class=\\"label\\">label1</div><style>
          .nav {
            display: flex;
            flex-direction: column;
            position: relative;
          }
          .label {
            cursor: ;
            font-weight: 400;
            white-space: nowrap;
            line-height: 22px;
            padding: 8px 16px;
            border-radius: 4px 4px 0px 0px;
            text-align: center;
            color: #444;
            background-color: #f1f3f6;
          }
          .label:hover {
            color: #0054fe;
            background-color: #f1f3f6;
          }
          .bottom {
            background-color: #0054fe;
            transition: all 200ms ease;
            opacity: 0;
            transform: scale(0.75);
            height: 4px;
            width: 100%;
          }
        </style></div></div><div class=\\"tab \\" role=\\"button\\"><div class=\\"nav\\"><div class=\\"label\\">label2</div><style>
          .nav {
            display: flex;
            flex-direction: column;
            position: relative;
          }
          .label {
            cursor: ;
            font-weight: 400;
            white-space: nowrap;
            line-height: 22px;
            padding: 8px 16px;
            border-radius: 4px 4px 0px 0px;
            text-align: center;
            color: #444;
            background-color: #f1f3f6;
          }
          .label:hover {
            color: #0054fe;
            background-color: #f1f3f6;
          }
          .bottom {
            background-color: #0054fe;
            transition: all 200ms ease;
            opacity: 0;
            transform: scale(0.75);
            height: 4px;
            width: 100%;
          }
        </style></div></div></header><div class=\\"content\\"></div><style>
            header {
              display: flex;
              flex-wrap: nowrap;
              overflow-x: auto;
              align-items: center;
              border-bottom: ;
            }
            ::-webkit-scrollbar {
              display: none;
            }
            .content {
              padding-top: 0.625rem;
            }

            .tab {
              cursor: pointer;
              outline: 0;
              text-transform: capitalize;
              font-size: 1rem;
              margin: 0 1px;
              user-select: none;
              display: flex;
              align-items: center;
              line-height: 1.25rem;
              position: relative;
              font-variant-numeric: tabular-nums;
            }
            .tab:first-of-type {
              margin-left: 0;
            }
          </style></div></div>"
`;
